### ISSUE TITLE
How to do continuous parallel plotting?

### ISSUE THREAD

**Issue by CharlesWiltgen** [maded2/plotng#28]
plotng is working great for me. I'm using it to do 3 plots in parallel with a 30m stagger. Over time, the plotting looks like this:

```
#####    #####    #####    #####
  #####    #####    #####    #####
    #####    #####    #####    #####
```

As that diagram shows, there's wasted time/resources at the beginning and end of each cycle where it's only doing 1 or 2 plots in parallel. Ideally, I'd like plotng to do this…

```
##### ##### ##### ##### ##### #####
  ##### ##### ##### ##### ##### #####
    ##### ##### ##### ##### ##### #####
```
…but I haven't figured out the combination of settings that would keep 3 plots going at all time. Is anyone doing this, and if so how did you configure it?

**Comment by maded2** [comment]
I think it really does depends on your hardware setup.  It will be good for you to share statistics and your config so that I can see if there anything to adjust.  In generate you configure a delay for 2 reason.

1.  Lack of CPU resource
2. Lack of IO bandwidth to target drive

so do you have only 1 target drive? is the target drive on USB? etc.

**Comment by CharlesWiltgen** [comment]
@maded2, I explained that terribly. Let me try again without the fancy ASCII art. 😂 (The only hardware bit that matters is that there's one temp drive, a hella-fast NVMe RAID 0 array that's never the bottleneck in this scenario.)

TLDR: The optimal plotting I'm looking to do with plotng is simply, "When the number of active plots drops below 3, start another plot." (3 is just the target in my case, presumably this would be configurable.)

With settings `"NumberOfParallelPlots": 3, "MaxActivePlotPerTarget": 3, "StaggeringDelay": 30`, here's what happens:

```
- Plot 1: 12:00am–06:00am
- Plot 2: 12:30am–06:30am
- Plot 3: 01:00am–07:00am

- Plot 4: 07:00am–01:00pm
- Plot 5: 07:30am–01:30pm
- Plot 6: 08:00am–02:00pm
```

Here's what I'd like to happen:

```
- Plot 1: 12:00am–06:00am
- Plot 2: 12:30am–06:30am
- Plot 3: 01:00am–07:00am

- Plot 4: 06:00am–12:00pm (because Plot 1 ended @ 06:00am)
- Plot 5: 06:30am–12:30pm (because Plot 2 ended @ 06:30am)
- Plot 6: 07:00am–01:00pm (because Plot 3 ended @ 07:00am)
```

With that more efficient behavior, this would allow that same machine to create at least one more plot per day.  I don't think plotng can do this now, or at least I couldn't find a way. I tried `"NumberOfParallelPlots": 999, "MaxActivePlotPerTemp": 3`, but plotng started more than 3 plots in parallel.

I hope that makes more sense!

**Comment by maded2** [comment]
so why not set StaggeringDelay to zero.  It will then start a new plot as soon as the last one finishes.

**Comment by maded2** [comment]
if there's no other issue I will close this ticket tomorrow
